docs: improve structure (#676)
open-energy-transition/open-tyndp: doc/sb.rst

.. SPDX-FileCopyrightText: Contributors to Open-TYNDP <https://github.com/open-energy-transition/open-tyndp>
..
.. SPDX-License-Identifier: CC-BY-4.0

######################
Scenario Building (SB)
######################

Scenario Building (SB) is the first phase of the Open-TYNDP workflow. Starting from raw

> ENTSO-E input datasets, it constructs a fully sector-coupled European energy system model

Reviewer: Fully sector-coupled seems a bit of an overstatement as we are not modelling industry, agriculture or heating and are only modelling transport partly. We could either leave out the "fully" or specify which sectors

Reviewer: ```suggestion
ENTSO-E input datasets, it constructs a sector-coupled European energy system model
```

> and solves a least-cost capacity expansion and dispatch optimisation. The solved network

Reviewer: As the we are currently only modelling NT and NT does not entail capacity expansion, we should make it more clear for which scenario what happens to avoid confusion.
```suggestion
and solves a least-cost capacity expansion (DE and GA scenarios only, see `Open-TYNDP scenarios <scenarios.html>`_) and dispatch optimisation. The solved network
```

`PyPSA-Eur <https://pypsa-eur.readthedocs.io/en/latest/>`_, inheriting its modelling
framework (optimisation structure, network representation, and sector-coupling capabilities)
while replacing specific inputs and assumptions to match TYNDP 2024 reference data.

Input Data
==========

All used input datasets are publicly available from the `TYNDP 2024 scenarios download page
<https://2024.entsos-tyndp-scenarios.eu/download/>`_. The diagram below shows how they
flow into Open-TYNDP and into the benchmarking process.

.. image:: img/tyndp/data_overview.png
    :align: center
    :alt: TYNDP 2024 Scenario Building data overview diagram

Public input data from ENTSO-E is used wherever available. Where publicly available data does
not match the fixed values observed in the Market Model output files, the output files are used
as the reference for fixed input assumptions. This applies strictly to exogenous variables that
are not part of the optimisation (H₂ demand profiles, reference grid topologies,
and generator maintenance profiles).

The following datasets are ingested and processed by dedicated ``build_tyndp_*`` Snakemake
rules:

> `PEMMDB 2.5 <https://2024-data.entsos-tyndp-scenarios.eu/files/scenarios-inputs/PEMMDB2.zip>`_

Reviewer: I noticed, that we are actually wrongfully referring to this as PEMMDB v2.4 in our repo. I will open a follow up PR to correct the version reference

    Installed generation and storage capacities, must-run constraints, and per-unit cost
    assumptions by country and technology. Also provides efficiency and variable O&M parameters
    for conventional thermal generation sourced from ERAA 2025.

`PECD 3.1 <https://2024-data.entsos-tyndp-scenarios.eu/files/scenarios-inputs/PECD.zip>`_
    Hourly capacity factor time series for wind (onshore and offshore) and solar PV, derived
    from ERA5 reanalysis data. Profiles are provided per climate year and TYNDP bidding zone.

`Hydro Inflows <https://2024-data.entsos-tyndp-scenarios.eu/files/scenarios-inputs/Hydro-Inflows.zip>`_
    Hourly inflow profiles for reservoir and run-of-river hydro plants, used to constrain

>     hydro dispatch across planning horizons.

Reviewer: ```suggestion
    hydro dispatch across planning horizons. Profiles are provided per climate year and TYNDP bidding zone.
```

    Optional extendable transmission and storage assets for the 2035 and 2040 planning horizons.

`Hydrogen <https://2024-data.entsos-tyndp-scenarios.eu/files/scenarios-inputs/Hydrogen.zip>`_
    Hydrogen storage parameters, steam methane reforming (SMR and SMR+CCS) capacities, and
    import pipeline assumptions.

`Supply Tool <https://2024-data.entsos-tyndp-scenarios.eu/files/scenarios-outputs/20240518-Supply-Tool.xlsm.zip>`_
    Methane demand and biomass potentials for energy carriers modelled in Open-TYNDP. This is
    a scenario output file from the TYNDP 2024 process used as a fixed input.

.. note::
    In the underlying PyPSA-Eur framework, RES profiles, hydro inflows, and demand time series
    can be derived directly from raw weather data without any additional model. The same applies
    to biomass potentials. Open-TYNDP currently uses the provided TYNDP input files for
    comparability with the established TYNDP 2024 methodology.

SB Workflow
===========

The SB workflow transforms raw ENTSO-E input datasets into a solved, sector-coupled PyPSA
network. The key stages are: integrating public input data, constructing the sector-coupled

> network, applying TYNDP-specific constraints, solving the capacity expansion optimisation and visualisating results.

Reviewer: ```suggestion
network, applying TYNDP-specific constraints, solving the capacity expansion optimisation, visualising results, and running the Open-TYNDP `benchmarking framework <benchmarking.html>`_.
```

Network Construction
--------------------

The PyPSA network is built at **bidding zone / country-level resolution** for the electricity and hydrogen

> sectors. Buses represent national-level aggregations; AC lines and DC links represent

Reviewer: ```suggestion
sectors respectively. Buses represent national-level aggregations; AC lines and DC links represent
```

cross-border interconnectors with capacities and impedances taken from the TYNDP Line Data.

Generator and storage components are attached to country buses using PEMMDB 2.5 capacity data.
Which carriers are **extendable** varies by scenario and planning horizon.

Sector Coupling
---------------

Open-TYNDP models the electricity and hydrogen sectors as fully coupled. For the Distributed

> Energy (DE) scenario, heating sector links are included in addition. Cross-sector components

Reviewer: ```suggestion
Energy (DE) and Global Ambition (GA) scenarios, heating sector links are included in addition (see `Open-TYNDP scenarios <scenarios.html>`_). Cross-sector components
```

include:

* **Electrolysers:** Convert electricity to hydrogen; capacity is either fixed per PEMMDB 2.5
  or left extendable depending on the scenario.
* **Fuel cells and back-pressure plants:** Reconvert hydrogen or gas to electricity.
* **Hydrogen network:** Dedicated H₂ pipelines between country buses are included for planning
  horizons from 2030 onward, using TYNDP Line Data. Zones with split H₂ grids (e.g. the
  Iberian Peninsula) are represented with separate H₂ buses.
* **SMR and SMR+CCS:** Grey and blue hydrogen production capacities from PEMMDB and TYNDP
  hydrogen datasets.
* **Demand-side electrification:** Where the scenario specifies it, electricity demand
  incorporates direct electrification of heat and transport end-uses.

> Capacity Optimisation

Reviewer: ```suggestion
Capacity and Dispatch Optimisation
```

---------------------

The SB optimisation minimises **total annualised system cost** (investment plus variable

> operating cost) subject to:

Reviewer: We could again make it clear that there is not necessarily a capacity expansion.
```suggestion
The SB optimisation minimises **total annualised system cost** (variable
operating plus optional investment cost) subject to:
```

Author: @daniel-rdt what about this?
```suggestion
The SB optimisation minimises **total annualised system cost** (variable operating cost, plus investment cost where capacity expansion is enabled) subject to:
```

Reviewer: Sounds good :)

* Hourly supply-demand balance for each carrier at every bus.
* Transmission capacity constraints, with optional extendability for candidate lines.
* CO₂ emission budgets derived from the TYNDP 2024 scenario pathway.
* Minimum and maximum generation constraints from PEMMDB, including must-run levels and
  scheduled maintenance outages.

> * Country-level annual hydrogen supply and demand balances.

Reviewer: ```suggestion
* Country-level annual hydrogen supply and demand balances.
* Capacity expansion constraints reflecting given trajectories in the case of the DE and GA scenario
```

The problem is formulated as a **linear programme (LP)** and solved with the configured
solver (HiGHS as default as an open-source alternative for lower
temporal resolution runs; other solvers like Gurobi/Mosek are also supported and recommended for high-resolution runs).

Each planning horizon is solved independently using the capacity assumptions fixed for that
horizon. The solved ``network.nc`` file is retrieved by the CBA workflow as
its starting point. See the `CBA documentation <cba.html>`_ for details.

Configuration
=============

SB settings are split across ``config/config.tyndp.yaml`` (run-level settings) and

> ``config/scenarios.tyndp.yaml`` (scenario-specific overrides).

Reviewer: ```suggestion
SB settings are split across ``config/config.tyndp.yaml`` (run-level settings) and
``config/scenarios.tyndp.yaml`` (scenario-specific overrides). You can refer to
the `configuration <configuration.html>`_ page for a more comprehensive list of available PyPSA-Eur
and Open-TYNDP configuration options.
```

Scenarios and Planning Horizons
--------------------------------

* ``scenario``: Selects the TYNDP 2024 scenario. Supported values are ``NT``
  (National Trends),  ``GA`` (Global Ambition) and ``DE`` (Distributed Energy).
* ``planning_horizons``: List of target years to solve (e.g. ``[2030, 2035, 2040, 2050]``). Each
  horizon is solved as an independent optimisation.
* ``run.name``: Identifies the run and determines the output directory; typically set to

>   the scenario/climate-year identifier (e.g. ``NT-cy2009``).

Reviewer: ```suggestion
  the scenario/climate-year identifier (e.g. ``NT-cy2009``).
  * ``launch_explorer``: Whether to launch the `PyPSA Explorer <https://github.com/open-energy-transition/pypsa-explorer>`_ after the model solve within the workflow. Default is ``True``.
```
